### ISSUE TITLE
OpenID Connect 1.0 Relying Party Role Support (OpenID Connect 1.0 Client)

### ISSUE THREAD

**Issue by the-maldridge** [authelia/authelia#2728]
## Feature Request
I would like to see Authelia support fuctioning as an OIDC client to use an external source of users for zero click authentication.

### Description

Currently it is not possible to do a truly single sign on experience because users will still see an Authelia sign in page.  I'd like my users to see the primary SSO panel they're used to seeing and then have them transparently auth to Authelia on their way through the proxy with OIDC like I do currently with Pomerium.  

### Use Case

This feature enables the truly "magic" feeling that happens with BeyondCorp architectures where the system just "knows" who you are without you seeing login screens or having to perform extra steps.  It reduces lower friction and can make adoption easier by making the change completely transparent.  Right now this is the only feature that is missing for me to fully adopt Authelia across my organization.

**Comment by pathmissing** [comment]
This would be an awesome feature! It would allow authelia to integrate better with existing single sign-on providers like Oka and OneLogin.

**Comment by janaka** [comment]
this feature would be a huge unlock.

**Comment by james-d-elliott** [comment]
Agreed. It's something we intend to eventually support. We're consistently making changes with this in mind such as the adoption of AMR.

**Comment by Redmega** [comment]
Google Auth would be amazing!

**Comment by IngussNeilands** [comment]
OoO, are we making progress?!? This ticket is alive since 2021 ...
+1 for Google Auth

**Comment by aymanbagabas** [comment]
Would love to see this implemented!

**Comment by james-d-elliott** [comment]
This is a feature I'm struggling to not jump right in and just do.

The thing holding me back is some prerequisites that I would like to accomplish before we start work on this, and the reason for this is to achieve a particular goal.

In 4.38 or 4.39 we adjusted how the authentication level was derived to use a scheme called Authentication Method References (AMR). This is a common system for cross-system communication of how the user authenticated.

Ideally Authelia will be able to define policies other than "one_factor" and "two_factor" which clearly define the makeup of the methods used and it automatically and dynamically prompts the user for the most appropriate method (or asks them to register it).

This AMR system is also part of OpenID Connect 1.0, and is one of the standard ID Token claims, and realistically I'd want admins to control (i.e. opt in) which providers they trust the AMR for and automatically copy into the Authelia session. This would allow a user to automatically satisfy the requirements for an app protected by Authelia based on how they authenticated with the provider they logged in with.

This also has a sister called Authentication Context Class Reference (ACR). ACR is basically an assurance profile that providers can offer, that clients can request in an authorization request, to ensure the user authenticates a certain way when they login. Often these ACR's link to a collection of AMR's that would satisfy the ACR. I'd like to offer this to OIDC clients of Authelia, completing the chain by allowing the client to proactively request higher levels of authentication.

---

The hard part here is making the choice of which authentication method to show in the uplift process on the UI, everything else is relatively simple, I think. We'll get there, though.
